docs/source/user-guide/nlp/nlp_properties.rst

```diff
@@ -0,0 +1,167 @@
+Formality*                      Formality of the text. Uses the s-nlp/roberta-base-formality-ranker model
+==============================  ==========
+
+*These properties are not calculated by default, as they may take a long time to calculate. To use them, pass
+``include_long_calculation_properties`` to the :class:`TextData.calculate_properties <deepchecks.nlp.TextData>` method.
+
+
+Label & Prediction Properties
+-------------------------------
+
+..note:
+    Not implemented yet
+
+
+Using Properties in Checks
+==========================
+
+Whether you are using the built-in properties or your own, the process of using them in the checks is the same.
+In order to use the properties of your text in a check, the properties should already be part of the ``TextData`` object.
+
+
+Calculating The Built-in Properties
+-----------------------------------
+
+In order to use the built-in properties, you must call the ``calculate_default_properties`` method of the ``TextData``
+object. This method will calculate the properties and add them to the :class:`TextData <deepchecks.nlp.TextData>` object.
+
+Example of calculating the built-in properties in order to use the TextPropertyOutliers check:
+In the following example, we will calculate the default properties in order to use the TextPropertyOutliers check:
+
+.. code-block:: python
+
+  from deepchecks.nlp.checks import TextPropertyOutliers
+  from deepchecks.nlp import TextData
+
+  # Initialize the TextData object
+  text_data = TextData(text)
+
+  # Calculate the default properties
+  text_data.calculate_default_properties()
+
+  # Run the check
+  TextPropertyOutliers().run(text_data)
+
+Including or Ignoring Properties
+#################################
+
+When calculating the properties, you can choose to include or exclude specific properties, by passing the
+``include_properties`` or ``ignore_properties`` parameters to the ``calculate_default_properties`` method.
+The parameters should be a list of the names of the properties to include or ignore. Note that only one of the
+parameters can be passed to the method.
+
+In the following example, we will calculate the built-in properties and ignore the ``Text Length`` property:
+
+.. code-block:: python
+
+  text_data.calculate_default_properties(ignore_properties=['Text Length'])
+
+
+Moreover, some properties are not calculated by default, as they may take a long time to calculate. In order to
+use them, pass ``include_long_calculation_properties`` to the ``calculate_default_properties`` method.
+
+In the following example, we will calculate the properties and include only the long calculation property "Toxicity":
+
+.. code-block:: python
+
+  text_data.calculate_default_properties(include_long_calculation_properties=True, include_properties=['Toxicity'])
+
+
+Using Your Own Properties
+-------------------------
+
+If you already have the properties calculated, there are 2 ways to use them:
```

Comment:
Missing the why - either you saved the deepchecks properties for this dataset somewhere to save time, or you calculated something smart of your own.